### PR TITLE
Resource markdown

### PR DESCRIPTION
**Pull request by myluki2000**
## Description of changes
* Implentation of parser etc. as defined by the ResourceMarkdown specification
* Remove the auth token from LoggedInUser again
* Change RequestHeaderUserProcessor such that it does not throw an exception if the JSON it parses contains a property the parser does not need to correctly fill all fields of LoggedInUser

  
## How has this been tested?
Please describe the test strategy you followed.

- [x] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test
    
## Checklist before requesting a review
- [ ] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/common/resource_markdown/ResourceMarkdownEntity.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+        // quite a complicated regex pattern to match a ResourceMarkdown link. For an explanation of this regex, see
+        // the specification document of ResourceMarkdown:
+        // https://gits-enpro.readthedocs.io/en/latest/dev-manuals/api-specifications/ResourceMarkdown.html
+        // The only thing added here are additional backslashes for
+        // Java string escaping, and two pairs of braces "( )" to create two capture groups, one for the resource type
+        // name (the string before the "/") and the other one for the UUID after the "/"
+        Pattern pattern = Pattern.compile(
+                "\\[\\[" +
+                        "([a-zA-Z]+)" +
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/common/resource_markdown/ResourceMarkdownEntity.java:50]: No Todo, just a hint: You can also name groups so you see what part they represent

**Review by PaulBredl — APPROVED** [review]

File: src/main/java/de/unistuttgart/iste/gits/common/resource_markdown/ResourceMarkdownEntity.java (hunk 1 above)
Comment (myluki2000) [reply src/main/java/de/unistuttgart/iste/gits/common/resource_markdown/ResourceMarkdownEntity.java:50]: Yea I know I just didn't want to make the pattern even longer
